cpp/src/arrow/compute/exec/expression.cc

```diff
@@ -1191,6 +1191,26 @@ Result<Expression> SimplifyWithGuarantee(Expression expr,
   return expr;
 }
 
+Result<Expression> RemoveNamedRefs(Expression src) {
+  if (!src.IsBound()) {
+    return Status::Invalid("RemoveNamedRefs called on unbound expression");
+  }
+  return ModifyExpression(
+      std::move(src),
+      [](Expression expr) {
```

Comment:
Nit: add parameter names for these callbacks?
```suggestion
      /*pre=*/ [](Expression expr) {
```

Reply:
Fixed.